storage/replica.go

```diff
@@ -841,6 +851,14 @@ func (r *Replica) applyRaftCommand(ctx context.Context, index uint64, originNode
 	} else {
 		// Update cached appliedIndex if we were able to set the applied index on disk.
 		atomic.StoreUint64(&r.appliedIndex, index)
+		if args.Method() == proto.TruncateLog {
+			ts := proto.RaftTruncatedState{}
+			_, err := engine.MVCCGetProto(r.rm.Engine(), keys.RaftTruncatedStateKey(r.Desc().RangeID),
+				proto.ZeroTimestamp, true, nil, &ts)
+			if err == nil {
```

Comment:
If this read fails, our truncated state will be incorrect, which is likely to lead to panics later on. We shouldn't let that pass silently. We should log a warning and call `setTruncatedState(nil)` in that case to invalidate the cache.